lxd/device/tpm.go

```diff
@@ -217,13 +217,24 @@ func (d *tpm) startVM() (*deviceConfig.RunConfig, error) {
 		},
 	}
 
-	proc, err := subprocess.NewProcess("swtpm", []string{"socket", "--tpm2", "--tpmstate", fmt.Sprintf("dir=%s", tpmDevPath), "--ctrl", fmt.Sprintf("type=unixio,path=%s", socketPath)}, "", "")
+	configSockDir, err := os.Open(tpmDevPath)
```

Comment:
This section could do with a comment explaining why the complexity is here. (i.e to overcome longer path names)